mkosi

```diff
@@ -2830,25 +2865,46 @@ def check_root():
     if os.getuid() != 0:
         die("Must be invoked as root.")
 
+def run_shell(args):
+    target = "--directory=" + args.output if args.output_format in (OutputFormat.directory, OutputFormat.subvolume) else "--image=" + args.output
+
+    cmdline = ["systemd-nspawn",
```

Comment:
I thought about those, and explicitly decided not to use them:

 1. `--as-pid2` is the right choice for most commands, but not so much for a shell, as shells generally have some basic support in place for being run as PID 1. Moreover, running them as PID 1 has the benefit that we can later on start the whole system, by doing "exec /usr/lib/systemd/system". Or in other words: if we run the shell this way it is fully equivalent to init=/bin/sh on a physical system, and that's a good thing:
2. Regarding `--quiet`: this is an interactive shell, hence I think we really should tell the user how he can exit it. I mean, there's a reason systemd-nspawn is by default not quiet: so that people know how to exit the thing, and this applies in this case as much as it applies for the general nspawn usecase.
3. Regarding `--register=no`: This is an interactive boot, and I think it's a really good thing that we can connetc from the outside to it. It's at least the way I usually work here: do something on the console of the container, and then get another shell in it via "machinectl shell" so that i can invoke strace or gdb or whatever else to supervise what I am doing there. By doing `--register=no` we'd take that away.

Or to say this all differntly: this is really about interactive shells, and the usecase is debugging: "please give me a quick shell in what i am working on here". The three options you suggested are the right choice for automatic build processes (and that's why we use them there), but not for these commands. I hope that makes sense?